src/ray/raylet/runtime_env_agent_client.cc

```diff
@@ -244,10 +244,10 @@ class SessionPool {
   // enqueue one.
   void remove_session_from_running(std::shared_ptr<Session> session) {
     running_sessions_.erase(session);
-    if (pending_sessions_.size() > 0) {
```

Comment:
Don't use `size` to decide empty to not, not all container guarantees O(1) `size` call.

Reply:
Is `empty()` guaranteed to be O(1)?

Reply:
For all the containers that I know, yes.

Reply:
A infamous example for O(N) `size` call is `std::list`:
- before C++11, there's no requirement from standard: https://en.cppreference.com/w/cpp/container/list/size
- quite a few discussions and debates: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=49561